boto3/docs/utils.py

```diff
@@ -11,11 +11,23 @@
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
 import inspect
+import re
 from botocore.compat import six
 
+TARGET_COMPONENT_RE = re.compile(r'[^.\[\]]+(?![^\[]*\])')
+
 
 def get_resource_ignore_params(params):
```

Comment:
I am sort of confused here. Can you give an example of chars appearing in literals?